### ISSUE TITLE
Improvements to RMSD.py script

### ISSUE THREAD

**Issue by vtlim** [MobleyLab/off-ffcompare#35]
Notes from our meeting 3-20-17
A lot of this is pseudo code and general comments so ask if you want more clarification.

```
# example usage
python RMSD.py --ref SMIRFF --compare GAFF,GAFF2,OPLS3,OPLS2005 --dir /path/containing/optimized/directories --output rmsd-comparison

# example directory layout
/path/containing/optimized/directories
/path/containing/optimized/directories/GAFF
/path/containing/optimized/directories/GAFF2
/path/containing/optimized/directories/etc.

_____________________________


def RMSD(Ref_mol2, Query_mol2, molName):

# update docstring
Ref_mol2: str, path/to/reference.mol2 file
Query_mol2: str, path/to/query.mol2 file
molName: str - name of the MOL2 file >>> rename to fName


1. update RMSD parameters to remove molSet and include directory
2. remove work cluster directory and replace with homedir
3. remove molNames
4. everything after line 74 should be gone by now
5. then add else: rms == -2
6. return rms

__________________________________

# help section for output, specify that it's the base output name for RMSD output file (base.txt), error output (base_err.txt), and -1 output file (base_neg1.txt).


# if main section
# parse four options

(opt, args) = parser.parse_args()


# parse list of force fields 
listFFs = opt.compare.split(',')

# get list of molecules
refMols = os.listdir(opt.dir + opt.ref)

# open output files using outputbase + some extension 


# for mol2_file in refMols
    # for queryFF in listFFs 
        ref_file = opt.dir + opt.ref + mol2_file
        query_file = opt.dir + queryFF + mol2_file
        molName = mol2_file.split('.')[0]
        # value = RMSD(ref_file, query_file, molName)
        lines 78-81 go here.. if
        lines 85-87 go here.. elif
         line 76 goes here.... else
         lines 88-90 (closing files)

# close files
```

**Comment by bannanc** [comment]
@vtlim heads up we made a few minor changes here.

**Comment by bannanc** [comment]
We would like to make the output have the following format:

```
# Reference Force Field: [Name]
# Molecule Set: [Name]
# MolName            GAFF     GAFF2   ... etc
DrugBank_a100     0.011    ...
```

Update script with these lines:
```
logfile.write("# Reference Force Field: %s \n" % opt.ref)
logfile.write("# Molecule Set Directory: %s \n" % opt.dir)

FF_string = "\t".join(listFFs) 
logfile.write("# MolName \t %s \n" % FF_string)

# for mol2_file in refMols
   rms_list = list()
    molName = mol2_file.split('.')[0]

   # for queryFF in listFFs 
        ref_file = opt.dir + opt.ref + mol2_file
        query_file = opt.dir + queryFF + mol2_file        
        # value = RMSD(ref_file, query_file, molName)
        if rms == -2: error file
        elif rms == -1: negative file
        else:
               rms_list.append("%.3e" % rms) 

    # For each molecule make a line of data
    rms_string = "\t".join(rms_list)
   "%20s\t %s" % (molName, rms_string)
# close files

```

**Comment by bannanc** [comment]
@nthi0625 above is how to update the output 

You can put `sorted` in front of any list so we should do that with the forcefields too, for example:

`refMols = sorted(os.listdir(opt.directory + opt.ref))`

**Comment by vtlim** [comment]
@bannanc - good to close?

**Comment by bannanc** [comment]
I think so

**Comment by bannanc** [comment]
These were all addressed in PR #47
